packages/test-base/build.gradle.kts

```diff
@@ -180,41 +176,28 @@ kotlin {
 }
 
 kotlin {
-    // define targets depending on the host platform (Apple or Intel)
     if(System.getProperty("os.arch") == "aarch64") {
         iosSimulatorArm64("ios")
         macosArm64("macos")
     } else if(System.getProperty("os.arch") == "x86_64") {
         iosX64("ios")
         macosX64("macos")
     }
-
     sourceSets {
-        val macosMain by getting
-        val macosTest by getting
-        getByName("iosMain") {
-            kotlin.srcDir("src/macosMain/kotlin")
-        }
-        getByName("iosTest") {
-            kotlin.srcDir("src/macosTest/kotlin")
+        val commonMain by getting
+        val commonTest by getting
+        val nativeDarwin by creating {
+            dependsOn(commonMain)
         }
-    }
-}
-
-// Needs running emulator
-tasks.named("iosTest") {
```

Comment:
I suppose we can still set the `iosDevice` property in Gradle properties/Env to request a specific simulator to align local/CI tests

Reply:
Maybe a good option to keep. Hooked the `iosDevice-property into the Gradle plugins infrastructure with https://github.com/realm/realm-kotlin/pull/1085/files#diff-ad7d5c068deb652e9518763651bed6874ac941679d187d260bec712efec3cf27R198